scripts/run_portserver_test.py

```diff
@@ -0,0 +1,454 @@
+
+"""Unit tests for scripts/run_portserver.py"""
+
+from __future__ import annotations
+
+import builtins
+import logging
+import os
+import socket
+import sys
+
+from core import utils
+from core.tests import test_utils
+from scripts import run_portserver
+
+from typing import Any, List
+
+
+class MockSocket:
+    server_closed = False
+    port: str|int = 8181
+
+    def setsockopt(self, *unused_args: Any) -> None: # pylint: disable=missing-docstring
+        pass
+
+    def bind(self, *unused_args: Any) -> None: # pylint: disable=missing-docstring
+        pass
+
+    def listen(self, *unused_args: Any) -> None: # pylint: disable=missing-docstring
+        pass
+
+    def getsockname(self, *unused_args: Any) -> List[Any]: # pylint: disable=missing-docstring
+        return ['Address', self.port]
+
+    def recv(self, *unused_args: Any) -> None: # pylint: disable=missing-docstring
+        pass
+
+    def sendall(self, *unused_args: Any) -> None: # pylint: disable=missing-docstring
+        pass
+
+    def shutdown(self, *unused_args: Any) -> None: # pylint: disable=missing-docstring
+        raise socket.error('Some error occurred.')
+
+    def close(self) -> None: # pylint: disable=missing-docstring
+        self.server_closed = True
+
+
+class MockServer:
+    def run(self) -> None: # pylint: disable=missing-docstring
+        pass
+
+    def close(self) -> None: # pylint: disable=missing-docstring
+        pass
+
+
+class CloudTransactionServicesTests(test_utils.GenericTestBase):
+    """Unit tests for scripts/run_portserver.py"""
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.terminal_logs: list[Any] = []
+        def mock_logging(*msgs: Any) -> None:
+            all_messages = [*msgs]
+            for msg in all_messages:
+                self.terminal_logs.append(msg)
+        self.swap_log = self.swap(logging, 'info', mock_logging)
+        self.terminal_err_logs: list[Any] = []
+        def mock_logging_err(*msgs: Any) -> None:
+            all_messages = [*msgs]
+            for msg in all_messages:
+                self.terminal_err_logs.append(msg)
+        self.swap_log_err = self.swap(logging, 'error', mock_logging_err)
+
+    def test_get_process_start_time_handles_ioerror(self) -> None:
+        def mock_open(*unused_args: Any, **unused_kwargs: Any) -> None:
+            raise IOError('File not found.')
+        pid = 12345
+
+        swap_open = self.swap_with_checks(
+            utils, 'open_file', mock_open,
+            expected_args=(('/proc/{}/stat'.format(pid), 'r'),))
+
+        with swap_open:
+            returned_time = run_portserver.get_process_start_time(pid)
+        self.assertEqual(returned_time, 0)
+
+    def test_get_process_start_time(self) -> None:
+        with open('dummy_file.txt', 'w', encoding='utf-8') as f:
+            f.write('A B C D E F G H I J K L M N O P Q R S T U 11 V')
+
+        dummy_file_object = open('dummy_file.txt', 'r', encoding='utf-8')
```

Comment:
You can use a stream to avoid creating the dummy file: https://docs.python.org/3/library/io.html. Ditto elsewhere

Reply:
Done